### PR TITLE
Use Auth SDK

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds Auth SDK usage, replacing old custom auth logic.

PR for 3.0: https://github.com/saleor/saleor-dashboard/pull/1593

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://jwt-expire-test-3-1.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4512 Use Auth SDK in Dashboard](https://app.clickup.com/t/2549495/SALEOR-4512)

**Review by Cloud11PL — APPROVED** [review]

File: src/components/AppLayout/AppChannelContext.tsx [hunk 1]
```diff
@@ -37,11 +37,18 @@ const isValidChannel = (
...
+
+export const AppChannelProvider: React.FC<AppChannelProviderProps> = ({
+  children,
+  onChannelChange
+}) => {
+  const { authenticated, authenticating } = useAuth();
   const [selectedChannel, setSelectedChannel] = useLocalStorage("channel", "");
   const { data: channelData, refetch } = useBaseChannelsList({
-    skip: !isAuthenticated
+    skip: !authenticated || authenticating
```
Comment (jwm0) [inline src/components/AppLayout/AppChannelContext.tsx:51]:
```suggestion
    skip: !authenticated
```
We only care if the user is `authenticated`, why do we need to check if it's in flight as well?

File: src/components/Shop/index.tsx [hunk 2]
```diff
@@ -14,10 +14,10 @@ type ShopContext = ShopInfo_shop;
 export const ShopContext = React.createContext<ShopContext>(undefined);
 
 export const ShopProvider: React.FC = ({ children }) => {
-  const { isAuthenticated } = useAuth();
+  const { authenticated, authenticating } = useAuth();
 
   return (
-    <TypedShopInfoQuery skip={!isAuthenticated}>
+    <TypedShopInfoQuery skip={!authenticated || authenticating}>
```
Comment (jwm0) [inline src/components/Shop/index.tsx:20]: Same here

File: src/index.tsx [hunk 3]
```diff
@@ -113,60 +114,60 @@ const apolloClient = new ApolloClient({
...
     }
   }),
-  link: authLink.concat(link)
+  link
+});
+
+const saleorClient = createSaleorClient({
+  apiUrl: API_URI,
+  channel: "",
+  autologin: true
```
Comment (jwm0) [inline src/index.tsx:123]: autologin is `true` by default

File: src/index.tsx [hunk 4]
```diff
@@ -113,60 +114,60 @@ const apolloClient = new ApolloClient({
...
   const { channel } = useAppChannel(false);
 
   const channelLoaded = typeof channel !== "undefined";
 
-  const homePageLoaded =
-    channelLoaded &&
-    isAuthenticated &&
-    !tokenAuthLoading &&
-    !tokenVerifyLoading;
+  const homePageLoaded = channelLoaded && authenticated && !authenticating;
```
Comment (jwm0) [inline src/index.tsx:168]: `authenticated` and `authenticating` are mutually exclusive. They can never both be `true`, so you might just want to skip the last check.

File: src/auth/hooks/useAuthProvider.ts [hunk 5]
```diff
@@ -1,58 +1,144 @@
...
+    setPassword
+  } = useAuth();
+  const { authenticated, authenticating } = useAuthState();
 
-  const saleorAuth = useSaleorAuthProvider({
-    authPlugin,
-    setAuthPlugin,
-    ...opts
-  });
+  const autologinPromise = useRef<Promise<any>>();
```
Comment (jwm0) [inline src/auth/hooks/useAuthProvider.ts:45]: Why are we passing a promise to context? If it's inside context it's only consumed by other react components, so why not use a state to check for resolve status?

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/auth/views/Login.tsx [hunk 6]
```diff
@@ -37,6 +38,10 @@ const LoginView: React.FC<LoginViewProps> = ({ params }) => {
   } = useQuery<AvailableExternalAuthentications>(
     availableExternalAuthentications
   );
+  const [
+    requestedExternalPluginId,
+    setRequestedExternalPluginId
+  ] = useLocalStorage("requestedExternalPluginId", null);
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/auth/views/Login.tsx:44]: SDK keeps information of pluginID which was used to log in, while here we are keeping pluginID which was used for redirection to external OAuth, thus we cannot use state as it won't be preserved after rdirection.

File: src/auth/utils.ts [hunk 7]
```diff
@@ -55,28 +18,15 @@ export const displayDemoMessage = (
...
-      }
-    } else {
-      logout();
-      notify({
-        status: "error",
-        text: intl.formatMessage(commonMessages.somethingWentWrong)
-      });
-    }
+    logout();
+    notify({
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/auth/utils.ts]: Ok, I've changed it to separate messages `sessionExpired` and `somethingWentWrong` respectively

File: src/auth/hooks/useAuthProvider.ts [hunk 8]
```diff
@@ -1,58 +1,144 @@
...
-    };
-  }
+  const handleLogin = async (email: string, password: string) => {
+    const result = await login({
+      email,
+      password
+    });
+
+    if (result?.data.tokenCreate.errors.length > 0) {
+      logout();
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/auth/hooks/useAuthProvider.ts]: Removed

File: src/auth/types.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
+import { User } from "@saleor/fragments/types/User";
+import {
+  AccountErrorFragment,
+  CreateToken,
+  ExternalAuthenticationUrl,
+  ExternalObtainAccessTokens,
+  MutationSetPasswordArgs,
+  SetPasswordMutation,
+  UserFragment
+} from "@saleor/sdk/dist/apollo/types";
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/auth/types.ts:10]: Updated, please check out

File: src/components/AppLayout/AppChannelContext.tsx [hunk 10]
```diff
@@ -54,6 +61,10 @@ export const AppChannelProvider: React.FC = ({ children }) => {
     }
   }, [channelData]);
 
+  React.useEffect(() => {
+    onChannelChange(selectedChannel);
```
Comment (jwm0) [inline src/components/AppLayout/AppChannelContext.tsx:65]: This method should ideally be accessible from a SDK hook. Something along the lines of:
```
const { channel, setChannel } = useSaleorConfig();
```

**Review by jwm0 — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**